acts

```diff
@@ -147,30 +147,49 @@ for dir in $backuptargets; do
     log_debug "message=\"Starting backup of $dir\""
     archive_starttime=$(date +%s)
     nicedirname=$(echo "$dir" | tr -d '/')
-    # Determine the archive type to create
-    if echo "$archives" | grep -E -q "^$hostname-yearly-$year-.+-$nicedirname$"; then
-        if echo "$archives" | grep -E -q "^$hostname-monthly-$year-$month-.+-$nicedirname$"; then
-            # There's a yearly and monthly backup already
-            archivetype="daily"
-        else
-            # There was a yearly but no monthly backup
-            archivetype="monthly"
-        fi
-    else
-        # There's no yearly backup
-        archivetype="yearly"
-    fi
-    log_verbose "archive-type type=$archivetype"
-    archivename="$hostname-$archivetype-$today-$nicedirname"
-    log_verbose "backup-start type=$archivetype dir=/$dir name=$archivename"
+    dailyarchive="$hostname-daily-$today-$nicedirname"
+    monthlyarchive="$hostname-monthly-$today-$nicedirname"
+    yearlyarchive="$hostname-yearly-$today-$nicedirname"
+    log_verbose 'archive-type type=daily'
+    log_verbose "backup-start type=daily dir=/$dir name=$dailyarchive"
     # Uncontrolled expansion is bad, but we have little choice. See https://github.com/koalaman/shellcheck/wiki/Sc2086
     # shellcheck disable=SC2086
-    if ! output="$($tarsnap -c -f "$archivename" -C / $tarsnapbackupoptions "$dir" 2>&1)"; then
-        log_message "backup-error type=$archivetype dir=/$dir output=\"$(echo $output | tr '\n' " ")\""
+    if ! output="$($tarsnap -c -f "$dailyarchive" -C / $tarsnapbackupoptions "$dir" 2>&1)"; then
+        log_message "backup-error type=daily dir=/$dir output=\"$(echo $output | tr '\n' " ")\""
         backuprc=1
     fi
     archive_endtime=$(date +%s)
-    log_verbose "backup-finish type=$archivetype dir=/$dir duration=$((archive_endtime - archive_starttime))s"
+    log_verbose "backup-finish type=daily dir=/$dir duration=$((archive_endtime - archive_starttime))s"
+
+    # Determine whether other backup levels need to be created too
+    # If so, copy them from $dailyarchive
+    if ! echo "$archives" | grep -E -q "^$hostname-yearly-$year-.+-$nicedirname$"; then
+        # No yearly backup
+        log_debug "message=\"Copying daily archive to yearly archive\""
+        archive_starttime=$(date +%s)
+        log_verbose "copy-start type=yearly dir=/$dir name=$yearlyarchive"
```

Comment:
if the copy process is relatively quick, are the `copy-start` messages needed? There will always be either a `copy-error` or `copy-finish` message

Reply:
I threw them in for consistency with the archive operation, but I am always a fan of simpler output. I’ll happily take them out.

Reply:
The copy time scales with the number of files in the archive, not the size of the archive. For archives with a huge number of files, the copy time will be less than the backup time, but still noticeable.